polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -5149,7 +5149,9 @@ void MLIRASTConsumer::run() {
     std::string mangledName;
     MLIRScanner::getMangledFuncName(mangledName, FD, CGM);
 
-    if (done.count(mangledName))
+    const std::pair<FunctionContext, std::string> doneKey(F.getContext(),
```

Comment:
A given function may be called in more than one context, so this change is trying to deal with that situation. I was wondering whether instead of a (context, string) pair we could use a `FunctionToEmit` type as the element of the set.
A FunctionToEmit object contains the declarator and the context so that should be sufficient. What do you think?

Reply:
I also thought about this because I agree it'd be the correct thing to do, but that would involve implementing comparison mechanisms (`operator<`) for `FunctionDecl`. In the end, we'd end-up doing something equivalent to comparing the mangled name, so I don't think that adds much value given that we'd have to recompute that every time.

Reply:
ok